app/models/exercise.rb

```diff
@@ -58,6 +58,10 @@ class Exercise < ApplicationRecord
     source: :contributor
 
   scope :sorted, -> { order(:position) }
+  scope :course, ->(track) { where(type: PracticeExercise.model_name.name) unless track.course? }
```

Comment:
Note: this is a conditional scope

Reply:
Can you explain what this does pls?

Reply:
Sure. If the track does _not_ have learning mode enabled, it will only return practice exercises. However, if the track _does_ have learning mode enabled, it'll not do anything so both concept and practice exercises are returned.